### PR TITLE
Clarify setup documentation

### PR DESCRIPTION
**Pull request by morganatlooker**
Made some clarifications and changed outdated material

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -28,24 +28,26 @@ Detailed information on how to interact with Lookerbot [can be found on Looker D
 
 #### Create a new bot in Slack
 
-1. Under "Customize Slack" > "Configure" > "Custom Integrations" select "Bots"
-2. Choose "Add Configuration"
+1. Go to http://yourdomain.slack.com/apps and choose "Bots".
```
Comment (wilg) [inline README.md]: I think there's actually a new proper way to create this and it's to create an internal "Slack App" at https://api.slack.com/apps

File: README.md [hunk 2]
```diff
@@ -28,27 +28,31 @@ Detailed information on how to interact with Lookerbot [can be found on Looker D
 
 #### Create a new bot in Slack
 
-1. Under "Customize Slack" > "Configure" > "Custom Integrations" select "Bots"
-2. Choose "Add Configuration"
+1. Go to https://api.slack.com/apps and create a new app. We use **looker** but it's up to you.
```
Comment (wilg) [inline README.md]: > We use **looker** but it's up to you.

This is confusing now

File: README.md [hunk 3]
```diff
@@ -28,27 +28,31 @@ Detailed information on how to interact with Lookerbot [can be found on Looker D
...
+2. Choose "Bot Users" and click "Add a bot user".
 3. Create a username for your Lookerbot. We use **@looker** but it's up to you.
-4. Choose an icon for Lookerbot. [Here's the icon we use](looker-bot-icon-512.png).
-5. Grab the API token from the settings page, you'll need this when you set up the bot server.
+4. Under settings, choose "Install App", then "Install App to Team" and authenticate.
+5. Copy the Bot User OAuth Access Token (you'll need this later)
+6. Under "Basic Information", you can add an icon and description for Lookerbot. [Here's the icon we use](looker-bot-icon-512.png).
+7. Note - make sure NOT to "distribute" the app.
```
Comment (wilg) [inline README.md]: note shouldn't be a "step 7" and we should explain why

Maybe instead just put this after:

```md
> By default, Slack Apps are internal to your team. Don't "distribute" your Slack App – that will make it available to all Slack users in the world.
```

File: README.md [hunk 4]
```diff
@@ -28,27 +28,31 @@ Detailed information on how to interact with Lookerbot [can be found on Looker D
...
 #### Manual Deployment
 
 The bot is a simple Node.js application. The application needs to be able to reach both your Looker instance's API and Slack's API. If you have a self-hosted instance of Looker, be sure to open up port 19999 (or your `core_port`) in order to accesss the Looker API.
 
+#### Environment Variables
+
 The bot is configured entirely via environment variables. You'll want to set up these variables:
 
-- `SLACK_API_KEY` (required) – Your Slack API key for the bot. You'll have gotten this when you created the bot user in the Slack settings.
+- `SLACK_API_KEY` (required) – Your Slack API key for the bot. You'll have gotten this when you created the bot user in the Slack settings. If you used the app to create a bot, you will use the Bot User OAuth Access Token in this field.
```
Comment (wilg) [inline README.md]: Let's just assume they used the app setup rather than explaining the old way and the new way.

File: README.md [hunk 5]
```diff
@@ -62,12 +66,20 @@ The bot is configured entirely via environment variables. You'll want to set up
 
 - `LOOKER_WEBHOOK_TOKEN` (optional) – The webhook validation token found in Looker's admin panel. This is only required if you're using the bot to send scheduled webhooks.
 
-- `SLACK_SLASH_COMMAND_TOKEN` (optional) – If you want to use slash commands with Lookerbot, provide the verification token from the slash command setup page so that the bot can verify the integrity of incoming slash commands.
+- `SLACK_SLASH_COMMAND_TOKEN` (optional) – If you want to use slash commands with Lookerbot, provide the verification token from the slash command setup page so that the bot can verify the integrity of incoming slash commands. If you used an app to set up the slash command, you can get the verification token in "Basic Information" in the app.
```
Comment (wilg) [inline README.md]: Let's just assume they used the app setup rather than explaining the old way and the new way.

File: README.md [hunk 6]
```diff
@@ -149,31 +161,30 @@ and use all the functionality.
...
-5. Set the URL to wherever you have your bot server hosted. The path to the slash command endpoint is `/slack/receive`, so if your bot is hosted at `https://example.com`, the URL would be `https://example.com/slack/receive`.
-6. You'll need to copy the token provided when you created the slash command and set the `SLACK_SLASH_COMMAND_TOKEN` variable with it for the bot to accept slash commands.
-
-Directions for creating slash commands [can be found in Looker Discourse](https://discourse.looker.com/t/using-lookerbot-for-slack/2302)
+4. Set the URL to wherever you have your bot server hosted (if you used Heroku to set up the server, this will be the unique app name that you set / generated) . The path to the slash command endpoint is `/slack/receive`, so if your server is at `https://example.com`, the URL would be `https://example.com/slack/receive`.
+5. Under settings, choose "Install App" again, then "Reinstall App" and authenticate.
+6. Under "Basic Information", grab the verification token. You'll use this to set the `SLACK_SLASH_COMMAND_TOKEN` evironment variable.
+7. Note - make sure NOT to "distribute" the app.
```
Comment (wilg) [inline README.md]: should only mention this once probably

File: README.md [hunk 7]
```diff
@@ -149,31 +161,30 @@ and use all the functionality.
...
+2. Choose "Slash Commands" and click "Create New Command".
 3. Create a command to use for the Looker bot. We use **/looker** but it's up to you.
-4. Choose an icon for the slash command responses. [Here's the icon we use](looker-bot-icon-512.png).
-5. Set the URL to wherever you have your bot server hosted. The path to the slash command endpoint is `/slack/receive`, so if your bot is hosted at `https://example.com`, the URL would be `https://example.com/slack/receive`.
-6. You'll need to copy the token provided when you created the slash command and set the `SLACK_SLASH_COMMAND_TOKEN` variable with it for the bot to accept slash commands.
-
-Directions for creating slash commands [can be found in Looker Discourse](https://discourse.looker.com/t/using-lookerbot-for-slack/2302)
+4. Set the URL to wherever you have your bot server hosted (if you used Heroku to set up the server, this will be the unique app name that you set / generated) . The path to the slash command endpoint is `/slack/receive`, so if your server is at `https://example.com`, the URL would be `https://example.com/slack/receive`.
+5. Under settings, choose "Install App" again, then "Reinstall App" and authenticate.
```
Comment (wilg) [inline README.md]: strange, didn't realize you needed to do this?

File: README.md [hunk 8]
```diff
@@ -149,31 +161,30 @@ and use all the functionality.
...
+4. Enter the webhook URL of the server you set up.
 
   - Post to public channels `/slack/post/channel/my-channel-name`
     - (Lookerbot will need to be invited to this channel to post in it.)
   - Post to private groups `/slack/post/group/my-channel-name`
     - (Lookerbot will need to be invited to this group to post in it.)
   - To direct message a user `/slack/post/dm/myusername`
 
-  These URLs are prefixed with the URL your bot. So, if yoru bot is hosted at `https://example.com` and you want to post to a channel called `data-science`, the URL would be `https://example.com/slack/post/channel/data-science`.
+  These URLs are prefixed with the URL of your server. (If you used the Heroku deployment, this will be the unique app name you specified/ generated). So, if your server is at `https://example.com` and you want to post to a channel called `data-science`, the URL would be `https://example.com/slack/post/channel/data-science`.
```
Comment (wilg) [inline README.md]: `specified/ generated` is awkward

File: README.md (hunk 2 above)
Comment (morganatlooker) [inline README.md]: changing this to "we call it looker"

File: README.md (hunk 7 above)
Comment (morganatlooker) [inline README.md]: I made a slash command and it popped up with a little message saying I needed to do that before changes would take effect

File: README.md [hunk 9]
```diff
@@ -28,27 +28,32 @@ Detailed information on how to interact with Lookerbot [can be found on Looker D
 
 #### Create a new bot in Slack
 
-1. Under "Customize Slack" > "Configure" > "Custom Integrations" select "Bots"
-2. Choose "Add Configuration"
+1. Go to https://api.slack.com/apps and create a new app. We call it **looker** but it's up to you.
```
Comment (wilg) [inline README.md]: should be capital L Looker

**Pull request merged**